package/gluon-mesh-vpn-wireguard/luasrc/lib/gluon/upgrade/400-mesh-vpn-wireguard

```diff
@@ -3,7 +3,7 @@
 local uci = require('simple-uci').cursor()
 local site = require 'gluon.site'
 
-local private_key = uci:get("network", 'wg_mesh', "private_key")
+local private_key = uci:get("network_gluon-old", 'wg_mesh', "private_key")
 
 if not private_key or not private_key:match("^" .. ("[%a%d+/]"):rep(42) .. "[AEIMQUYcgkosw480]=$") then
```

Comment:
The wireguard private key is lost if the firmware is upgraded to a firmware without the package `gluon-mesh-vpn-wireguard`. I am not yet sure, if this is a problem for us, but I thought, it might be worth noting.

Reply:
Interesting. I don't think it's critical to fix this right now - after all, an upgrade on a node that never had `gluon-mesh-vpn-wireguard` in the first place would lead to a similar situation. A possible solution would be to move the private key to `/etc/config/gluon`.

Reply:
> an upgrade on a node that never had gluon-mesh-vpn-wireguard in the first place would lead to a similar situation.

I do not understand? Of course, a node that never had `gluon-mesh-vpn-wireguard` does not have a wireguard key. So there is nothing to loose, but in which regard is this similar to a node that looses it's key?

Reply:
Well, a site going from firmwares (wireguard) -> (no wireguard) -> (wireguard) would be very unusual (as Gluon generally doesn't support downgrades), but if such a thing actually happened, the site would already need to have some solution in place for nodes that were first installed with the (no wireguard) firmware - either by requiring node owners to return to the config mode to register their key, or by having some kind of auto registration.

So this would just mean that every node going through the (no wireguard) -> (wireguard) migration will have to go through the same steps, regardless of whether it had a firmware with wireguard in the past or not. I think that's good enough for this unusual case.

Reply:
Ok.

Reply:
In general, I would like to see this moved to /etc/config/gluon.

As an upgrade with `sysupgrade` happens silently (without a config mode that shows you your new vpn key) this may remain unnoticed by the user and the user might start to wonder, why his vpn doesn't work anymore.

I agree, that this will not happen often most likely, but it might happen. As the fix is easy, I would suggest to go for it.

Reply:
Well, cleaning up behind packages that aren't installed anymore could also be seen as a feature - actually one that I'd like to have in all packages eventually.